### PR TITLE
chore: release 0.39.0

### PR DESCRIPTION
**Pull request by thbcmlowk**
image

https://substra-documentation--438.org.readthedocs.build/en/438/additional/release.html

### CODE REVIEW

File: docs/source/examples/substra_core/diabetes_example/assets/functions/local_second_order_computation/Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 FROM python:3.9-slim
 
 # install dependencies
-RUN pip3 install 'pandas<2.0' 'numpy<2.0' 'substratools==0.22.0a2'
+RUN pip3 install pandas 'numpy<2.0' 'scikit-learn==1.5.0' substratools
```
Comment (SdgJlbl) [inline docs/source/examples/substra_core/diabetes_example/assets/functions/local_second_order_computation/Dockerfile:5]:
```suggestion
RUN pip3 install pandas 'numpy<2.0' substratools
```
```suggestion
RUN pip3 install pandas 'numpy<2.0' 'scikit-learn==1.5.0' substratools
```

File: docs/source/additional/release.rst [hunk 2]
```diff
@@ -31,6 +31,24 @@ This is an overview of the main changes, please have a look at the changelog of
 - `backend changelog <https://github.com/Substra/substra-backend/blob/main/CHANGELOG.md>`__
 - `orchestrator changelog <https://github.com/Substra/orchestrator/blob/main/CHANGELOG.md>`__
 
+Substra 0.39.0 --- 2024-09-13
```
Comment (guilhem-barthes) [inline docs/source/additional/release.rst:34]: Add warning to stop support python 3.9

File: docs/source/additional/release.rst [hunk 3]
```diff
@@ -31,6 +31,24 @@ This is an overview of the main changes, please have a look at the changelog of
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**All**
+
+- The release 0.39.0 focuses on using GPU and on-demand GPU. The main visible change is the introduction of the new fields in susbtra-backend charts `worker.computePod.affinity`, `worker.computePod.nodeSelector` and `worker.computePod.tolerations`, allowing the use of node pools spawned on-demand with specific capabilities (like with a GPU).
+
+Other changes include:
+- Allow re-use of secrets (instead of runtime generation) in substra-backend
+- Reduced build time by downloading functions only if not present on the current backend
```
Comment (thbcmlowk) [inline docs/source/additional/release.rst:43]: I feel like this has been done 2 years ago 😅

File: docs/source/examples/substra_core/diabetes_example/assets/functions/local_second_order_computation/Dockerfile [hunk 4]
```diff
@@ -2,7 +2,7 @@
 FROM python:3.9-slim
 
 # install dependencies
-RUN pip3 install 'pandas<2.0' 'numpy<2.0' 'substratools==0.22.0a2'
+RUN pip3 install pandas 'numpy<2.0' substratools
```
Comment (thbcmlowk) [inline docs/source/examples/substra_core/diabetes_example/assets/functions/local_second_order_computation/Dockerfile:5]:
```suggestion
RUN pip3 install pandas<2.0 'numpy<2.0' substratools
```

File: docs/source/examples/substra_core/diabetes_example/assets/functions/local_second_order_computation/Dockerfile (hunk 1 above)
Comment (thbcmlowk) [reply docs/source/examples/substra_core/diabetes_example/assets/functions/local_second_order_computation/Dockerfile]: Thank you!

File: docs/source/additional/release.rst (hunk 2 above)
Comment (thbcmlowk) [reply docs/source/additional/release.rst:34]: Aye aye

**Review by SdgJlbl — APPROVED** [review]
Thanks for the release 🚀

**Pull request merged**